app/workers/udacity_course_worker.rb

```diff
@@ -73,11 +73,14 @@ def handle_response_data(response_data)
   end
 
   def calculate_duration(value, unit)
+    return nil if value == 0 && unit == ''
```

Comment:
Why are you using `&&` (instead of `||`)? What if the unit is different than these three we expect (or what if it is `null`)?